apps/web/app/(app)/environments/[environmentId]/people/[personId]/page.tsx

```diff
@@ -1,10 +1,94 @@
-import PersonDetails from "./PersonDetails";
+import { getPersonWithAttributeClasses } from "@formbricks/lib/services/person";
+import { getResponsesByPersonId } from "@formbricks/lib/services/response";
+import { getSessionWithActionsOfPerson } from "@formbricks/lib/services/session";
+import { getDisplaysOfPerson } from "@formbricks/lib/services/displays";
+import { getSurvey } from "@formbricks/lib/services/survey";
+import AttributesSection from "@/app/(app)/environments/[environmentId]/people/[personId]/AttributesSection";
+import ActivitySection from "@/app/(app)/environments/[environmentId]/people/[personId]/ActivitySection";
+import HeadingSection from "@/app/(app)/environments/[environmentId]/people/[personId]/HeadingSection";
+import ResponseSection from "@/app/(app)/environments/[environmentId]/people/[personId]/ResponseSection";
+import { TResponseWithSurveyData } from "@formbricks/types/v1/responses";
+
+export default async function PersonPage({ params }) {
+  let [personWithAttributes, displays, sessionsWithActions, responses] = await Promise.all([
+    getPersonWithAttributeClasses(params.personId),
+    getDisplaysOfPerson(params.personId),
+    getSessionWithActionsOfPerson(params.personId),
+    getResponsesByPersonId(params.personId),
+  ]);
+
+  if (!personWithAttributes) {
+    throw new Error("No such person found");
+  }
+  sessionsWithActions = sessionsWithActions ?? [];
+  displays = displays ?? [];
+
+  const surveyIds = responses?.map((response) => response.surveyId) || [];
+  const surveys = await Promise.all(surveyIds.map((surveyIdx) => getSurvey(surveyIdx)));
```

Comment:
I think for the database it could be faster to call `getSurveys()` because it can list all surveys with just one call instead of making multiple database calls for the surveys needed.

Reply:
done, added a check to only fetch the surveys in the environment if the user atleast has one response, this would help reduce the extra DB calls to fetch the surveys even when a user has no responses!